homeassistant/components/withings/coordinator.py

```diff
@@ -18,12 +18,12 @@
     aggregate_measurements,
 )
 
-from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryAuthFailed
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 from homeassistant.util import dt as dt_util
 
+from . import WithingsConfigEntry
```

Comment:
This import should be moved into an `if TYPE_CHECKING` block to prevent a circular import.

_This file also needs the `from __future__ import annotations` import to support forward references without quotes._

Reply:
> "support forward references without quotes"

Reply:
How does that work?

Reply:
> > "support forward references without quotes"
> 
> This wasn't updated

The future import was added. Otherwise the move to the `if TYPE_CHECKING` block wouldn't have worked.

Reply:
Sorry - I just took a fresh look.
It seems that you can drop the quotes only if you move the type alias AFTER the class declaration for `WithingsData`

Reply:
Ah, I had misunderstood the original comment.
I spotted it whilst working on #116450